libusb/os/windows_winusb.c

```diff
@@ -1826,7 +1808,24 @@ static int winusb_get_device_list(struct libusb_context *ctx, struct discovered_
 				priv->sub_api = sub_api;
 				switch (api) {
 				case USB_API_COMPOSITE:
+					break;
 				case USB_API_HUB:
+					parent_dev = get_ancestor(ctx, dev_info_data.DevInst, NULL);
+					if (parent_dev == NULL) {
+
+						DWORD hub_port_nr;
```

Comment:
Declarations just at the start of the block, and an empty line /after/,

Reply:
OK, changed with last commit

Reply:
Actually if the variable is local to the block, it can be declared at the start of this block, but with empty line after and not before.

Reply:
Ah, ok, sorry I misunderstood your comment. Do you prefer I declare it at the start of the block? I admit I didn't look in the whole code into details for block-local variable, but it seems the general style is to put everything at the top...

Reply:
OK, if the general style is to put everything at the top of the function, it is fine like that. In my own code I prefer to give variables as local scope as possible, but it is more important here to be consistent.

Reply:
@tormodvolden : OK, after looking at more code, I found example of exactly what you said I shall do: "block-local variable declared at the start of this block, but with empty line after and not before.", so I did it back that way (I prefer it as well and is compliant with style used around in this file, even if there is mix of styles). 

Sorry for the back-and-forth.

However I also found a lot of variables that are declared at the top of the function but could be declared locally in block. I think we should maybe plan a PR for fixing all this since, like you, I think this is better for many reason (one being that it is much less error-prone in my opinion). But of course the problem is all the conflicts in pending that this would potentially create...